package/src/components/ChannelPreview/hooks/__tests__/useChannelPreviewDisplayName.test.js

```diff
@@ -0,0 +1,139 @@
+  let chatClient;
+  let channel;
+  const initializeChannel = async (c) => {
+    useMockedApis(chatClient, [getOrCreateChannelApi(c)]);
+
+    channel = chatClient.channel('messaging');
+
+    await channel.watch();
+  };
+
+  beforeEach(async () => {
+    chatClient = await getTestClientWithUser(clientUser);
+  });
+
+  afterEach(() => {
+    channel = null;
+  });
+
+  it('will return a channel display name', async () => {
+    const channelName = 'okechukwu';
+    const characterLength = 15;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const TestComponent = () => {
+      const channelDisplayName = useChannelPreviewDisplayName(channel, characterLength);
+
+      return <Text>{channelDisplayName}</Text>;
+    };
+
+    const { getByText } = render(<TestComponent />);
+
+    await waitFor(() => {
+      expect(getByText(channelName)).toBeTruthy();
+    });
+  });
+
+  it('will return the same channelName when channelName length is less than characterLength', async () => {
+    const channelName = 'okechukwu';
+    const characterLength = 15;
+    const currentUserId = chatClient.userID;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const TestComponent = () => {
+      const channelDisplayName = useChannelPreviewDisplayName(channel, characterLength);
+
+      return <Text>{channelDisplayName}</Text>;
+    };
+
+    render(<TestComponent />);
+
+    const displayName = getChannelPreviewDisplayName({
+      channelName,
+      currentUserId,
+      maxCharacterLength: characterLength,
+      members: channel['state']['members'],
+    });
+
+    expect(displayName).toEqual(channelName);
+  });
+
+  it('will return the same channelName length as the characterLength', async () => {
+    const channelName = 'okechukwu nwagba martin';
+    const characterLength = 15;
+    const currentUserId = chatClient.userID;
+    await initializeChannel(
+      generateChannelResponse({
+        channel: {
+          name: channelName,
+        },
+      }),
+    );
+
+    const displayName = getChannelPreviewDisplayName({
+      channelName,
+      currentUserId,
+      maxCharacterLength: characterLength,
+      members: channel['state']['members'],
+    });
+
+    expect(displayName.length).toEqual(channelName.length);
+  });
+
+  test.each([
+    [15, GROUP_CHANNEL_MOCK, 'okechukwu nwagba', 'ben, nick,... +2'],
+    [15, CHANNEL_WITH_ONE_MEMBER_MOCK, 'okechukwu nwagba', 'okechukwu nwagb...'],
+    [15, CHANNEL_WITH_ONE_MEMBER_AND_EMPTY_USER_MOCK, 'okechukwu nwagba', 'Unknown User...'],
+  ])(
+    'getChannelPreviewDisplayName(%i, %p, %s) result in %s',
+    async (characterLength, members, currentUserId, expected) => {
+      await initializeChannel(
+        generateChannelResponse({
+          channel: {},
+        }),
+      );
```

Comment:
In this specific case, I think you can just drop the `initializeChannel` call altogether since you seem like you're doing what I mention above in this test